code/script.js

```diff
@@ -220,106 +226,154 @@ const generateBoard = () => {
+      );
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  } else if (category === 'hair') {
+      if (keep) {
+      alert(
+        `Yes, the person has ${value} hair! Keep all the people with ${value} hair.`
+        );
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
+    } else {
+      alert(
+        `No, the person does not have ${value} hair! Remove all people with ${value} hair.`
+        );
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+    }
+  } else if (category === 'eyes') {
+      if (keep) {
+      alert (
+        `Yes, the person has ${value} eyes! Keep all the people with ${value} eyes.`
+        );
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+       `No, the person does not have ${value} eyes! Remove all people with ${value} eyes.`
+         );
+         charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+      }
+    }
+    generateBoard(charactersInPlay);
+};
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  const confirmGuess = confirm(
+    `Are you sure you want to guess on ${personToConfirm} ?`
+    )
+    if (confirmGuess === true) {
+      checkMyGuess(personToConfirm) // Invoking the checkMyGuess function.
+    }
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  // Checks if the personToCheck is the same as the secret person's name
+  if (personToCheck === secret.name) {
+    // Play sound effect
+    const audio = new Audio(`images/success.mp3`)
+    audio.play() 
+
+    winOrLose.style.display = "flex"; // Show the win or lose section
```

Comment:
The code on lines 351 and 352 where you hide the game board and show the win or lose section comes again in lines 359 and 360. Since they happen in both "if" and "else", this code can be moved out from the if/else statement, just underneath where you declare your checkMyGuess-variable. By doing this, the code would be more DRY:)